First draft of combination indicator documentation (#148)
cmu-delphi/delphi-epidata: docs/api/covidcast-signals/indicator-combination.md

  `smoothed_adj_cli`. This signal is deprecated and is no longer updated as of
  May 28, 2020.

### Estimation

Let $$x_{rs}(t)$$ denote the value of sensor $$s$$ on day $$t$$ in region $$r$$
(*not* necessarily mean centered). We aim to construct a single scalar latent
variable each day for every region $$r$$, denoted by $$z_{r}(t)$$, that can
simultaneously reconstruct all other reported sensors. We call this scalar
summary the *combined* indicator signal. Below we describe the method used to
compute the combined signal. For notational brevity, we fix a time $$t$$ and
drop the corresponding index.

#### Optimization Objective

At each time $$t$$ (which in our case has resolution of a day), given a set
$$\mathcal{S}$$ of sensors (with total $$S$$ sensors) and a set $$\mathcal{R}$$

> of regions (with total $$R$$ regions), we aim to find a combined indicator that

Reviewer: ```suggestion
we construct
```

> of regions (with total $$R$$ regions), we aim to find a combined indicator that

Reviewer: ```suggestion
of regions (with total $$R$$ regions), we construct a combined indicator that
```

best reconstructs all sensor observations after being passed through a learned
sensor-specific transformation $$g_{s}(\cdot)$$. That is, for each time $$t$$, we

> aim to minimize the sum of squared reconstruction error:

Reviewer: ```suggestion
minimize the sum of squared reconstruction error:
```

$$
\sum_{s \in \mathcal{S}} \sum_{r \in \mathcal{R}} (x_{rs} - g_{s}(z_{r}))^2,
$$

over combined indicator values $$z = z_{1:R}$$ and candidate transformations $$g
= g_{1:S}$$. To solve the optimization problem, we constrain $$z$$ and $$g$$ in
specific ways as explained next.

#### Optimization Constraints

We constrain the sensor-specific transformation $$g_{s}$$ to be a linear

> function $$g_{s}(x) = a_{s} x$$. Then, the transformations $$g$$ are simply

Reviewer: ```suggestion
function such that $$g_{s}(x) = a_{s} x$$. Then, the transformations $$g$$ are simply
```

matrix factorization problem as follows:

$$
\min_{z \in\mathbb{R}^{R}: z \ge 0, a\in\mathbb{R}^S : a \ge 0} \|X - z a^\top\|_F^2.
$$

### Data Preprocessing

#### Scaling of Data

Since different sensor values are on different scales, we perform global column
scaling. Before approximating the matrix $$X$$, we scale each column by the root
mean square observed entry of that sensor over all of time (so that the same
scaling is applied each day).

Note that one might consider local column scaling before approximating the
matrix $$X$$, where we scale each column by its root mean square observed entry
each so that each sensor has (observed) second moment equal to 1 (each day). But
this suffers from the issue that the locally scaled time series for a given
sensor and region can look quite different from the unscaled time series. In
particular, increasing trends can be replaced with decreasing trends and, as a
result, the derived combined indicator may look quite distinct from any unscaled
sensor. This can be avoided with global column scaling.

#### Lags and Sporadic Missingness

The matrix $$X$$ is not necessarily complete and we may have entries missing.

> Several forms of missingness arise in our data.
> 
> * On certain days, all observations of a given sensor are missing due to release
>   lag. For example, Doctor Visits is released several days late.
> * On any given day, different sensors are observed in different regions.
> * For any given region and sensor, the sensor may be available on some days but
>   not others due to sample size cutoffs.

Reviewer: ```suggestion
Several forms of missingness arise in our data. On certain days, all observations of a given sensor are missing due to release lag.  For example, Doctor Visits is released several days late. Also, for any given region and sensor, a sensor may be available on some days but not others due to sample size cutoffs. Additionally, on any given day, different sensors are observed in different regions.
```

To ensure that our combined indicator value has comparable scaling over time and
is free from erratic jumps that are just due to missingness, we use the

> following strategies:
> 
> * *Lag imputation*: If a sensor is missing for all regions on a given day, we
>   copy all observations from the last day on which any observation was available
>   for that sensor.
> * *Recent imputation*: If $$x_{rs}(t)$$ is missing but at least one of
>   $$x_{rs}(t-1), x_{rs}(t-2), \dots, x_{rs}(t-T)$$ is observed, impute
>   $$x_{rs}(t)$$ with the most recent of $$x_{rs}(t-1), x_{rs}(t-2), \dots,
>   x_{rs}(t-L)$$. We limit $$T$$ to be 7 days.

Reviewer: ```suggestion
following imputation strategies:
*lag imputatation*, where if a sensor is missing for all regions on a given day, we copy all observations from the last day on which any observation was available for that sensor;
*recent imputation*, where if a sensor value if missing on a given day is missing but at least one of past $T$ values is observed, we impute it with the most recent value. We limit $T$ to be 7 days.
```

#### Persistent Missingness

Even with the above imputation strategies, we still have issues that some
sensors are never available in a given region. The result is that combined
indicator values for that region that may be on a completely different scale
from values in other regions with additional observed sensors. This can only be

> overcome by regularizing / pooling information across space. Note that a very

Reviewer: ```suggestion
overcome by regularizing or pooling information across space. Note that a very
```

similar problem occurs when a sensor is unavailable for a very long period of
time (so long that recent imputation is inadvisable and avoided by setting $$T =
7$$ days).

> We deal with this problem by *geographic imputation* where we impute values from
> regions that share a higher level of aggregation (e.g., the mean observed score
> in an MSA / state), or by imputing values from megacounties (since the counties
> in question are missing and hence should be reflected in the rest of state
> estimate). The order in which we try to perform geographic imputations is first
> attempting to impute from observed values from megacounties, followed by median
> observed values in the geographic hierarchy (county, MSA, and state), followed
> finally by the median observed value over the entire country if all the previous
> imputation attempts fail. This imputation order is chosen among different
> options by evaluating their effectiveness to mimic the actual observed values in
> validation experiments.

Reviewer: ```suggestion
We deal with this problem by *geographic imputation*, where we impute values from
regions that share a higher level of aggregation (e.g., the median observed score
in an MSA or state), or by imputing values from megacounties (since the counties
in question are missing and hence should be reflected in the rest of state
estimate). The order in which we look to perform geographic imputations is
observed values from megacounties, followed by median
observed values in the geographic hierarchy (county, MSA, state, or country).
We chose this imputation sequence among different options by evaluating 
their effectiveness to mimic the actual observed sensor values in validation experiments.
```
